### ISSUE TITLE
Strawberry creating duplicate tracks when using "move to collection"

### ISSUE THREAD

**Issue by manofsticks** [strawberrymusicplayer/strawberry#1341]
This appears to be the same error as seen in issue 902, but that one's closed and I have a lot more details, so I opened this new ticket. If you'd prefer to re-open that issue and have me move this information in, let me know. 

-In short: When I hit "move to collection" some tracks appear duplicated in the "Collection" tab. Goes away after I "organize" the album.

I made a youtube video showing the error in realtime if that's easier. 

https://www.youtube.com/watch?v=CQlU7elMxTE

Details:

-Fully updated Arch Linux
-Strawberry 1.0.22 (although it's been happening for a while for me, last few versions)

-Happens with .opus and .ogg files at minimum, I believe it's happened with MP3 in the past as well
-If it happens to one track, will happen to all subsequent tracks in the album
-Anecdotally seems to usually happen starting at track 10 or 11. Sometimes later, ~but don't believe I've ever seen it earlier than track 10, making me think something about the double digit renaming causes it~.
-Happens almost every album for me, so I'm guessing it's related to something in my workflow/auto track renaming.
-Hitting "organize" will usually fix it, but not always. I've even seen it duplicate the files again, causing 4 instances of the same track, but that is rarer and I cannot consistently reproduce it.

**Comment by manofsticks** [comment]
After making this report, I've actually hit an instance where a "less than 10" track has this issue, so I guess disregard that detail.

**Comment by jonaski** [comment]
I can't reproduce this but check if 1ff2bfd390703026d3f4315aa10f8d2b0d950c91 fixes it.

**Comment by manofsticks** [comment]
This unfortunately did not fix it. I tried several times and got inconsistent results on number of duplicates, so it doesn't look like the change made things worse, I just happened to take a screen recording on one of the worse ones. Video below, shows the console output too, which doesn't show anything to me that would help (only shows each file processing once).

https://www.youtube.com/watch?v=ZU0lwMpkYp0

Steps I followed in case I did something wrong in my testing:

Compiled from source, verified the change from https://github.com/strawberrymusicplayer/strawberry/commit/1ff2bfd390703026d3f4315aa10f8d2b0d950c91 was included (looks like it's already in the master branch, so I didn't make any changes after the git clone).

Only make warning I got was

```
CMake Warning at /usr/lib/cmake/Qt6/Qt6Config.cmake:223 (message):
  Failed to find optional Qt component "LinguistTools".

  Expected Config file at
  "/usr/lib/cmake/Qt6LinguistTools/Qt6LinguistToolsConfig.cmake" does NOT
  exist

Call Stack (most recent call first):
  CMakeLists.txt:198 (find_package)
```
Which I assume won't impact testing.

md5sum dd15038563b33cd2fb341bb640ba5f7d

**Comment by manofsticks** [comment]
This unfortunately did not fix it. Video below, shows the console output too.

https://youtu.be/_OPZ2WDFzWI

Steps I followed in case I did something wrong in my testing:

Compiled from source, verified the change from https://github.com/strawberrymusicplayer/strawberry/commit/1ff2bfd390703026d3f4315aa10f8d2b0d950c91 was included (looks like it's already in the master branch, so I didn't make any changes after the git clone).

Only make warning I got was

```
CMake Warning at /usr/lib/cmake/Qt6/Qt6Config.cmake:223 (message):
  Failed to find optional Qt component "LinguistTools".

  Expected Config file at
  "/usr/lib/cmake/Qt6LinguistTools/Qt6LinguistToolsConfig.cmake" does NOT
  exist

Call Stack (most recent call first):
  CMakeLists.txt:198 (find_package)
```
Which I assume won't impact testing.

md5sum of the end build: dd15038563b33cd2fb341bb640ba5f7d

**Comment by jonaski** [comment]
As far as I can tell from your logs, there is no indication that anything is updated or added to the database, which is really strange considering that you are getting all those duplicates, is this the complete logs?

I also notice that you are moving files from /media/MyMedia/MUSIC to /media/MyMedia/music, is /media/MyMedia/MUSIC a completely different path not part of the collection in any way through symlinks?

If anything is updated in the database, it would log something like this prefixed with CollectionWatcher.:

- <filename> is new
- unavailable song <filename restored.
- Song <filename> changed.

But the only thing I see from your logs related to CollectionWatcher is.:
` Subdir "/media/MyMedia/MUSIC/Kendrick Lamar" changed under directory `.

So when it's not logging anything more after "Checking for valid file", it means the mtime matches so there is no need to update the database.

There is only one other place where the song can be added/updated and that's the organize feature itself, but after my fix it should only update a song if the song that is organized already is a collection song, which it shouldn't be.

**Comment by manofsticks** [comment]
The "from" path is "/media/MyMedia/music to add/!UPGRADE" in this instance, a completely separate path. Not sure if it's an issue with the logs, or my terminal emulator that's removing the escape character before a space. But there is no "media/MyMedia/music" 

I believe I posted the logs from the moment I hit "move to collection" until completion, apologies if you needed any startup logs (or if I messed up).

Please see the below logs from a new run, everything from start, verifying the tracks are not there, "Move to Collection", then checking and seeing the duplicates. As part of this log I am not performing "organize files" after I move to the collection. With this instance, I saw 2 instances of each track in the Collection.

`[stickman@Sweden /home/stickman/Desktop/Strawberry_Test/strawberry/build]$ ./strawberry
15:39:26.221 INFO  main:180                         Strawberry 1.0.22-8-g55b1d34f Qt "6.6.1"
15:39:26.221 INFO  main:181                         "Arch Linux unknown - (linux 6.6.8-arch1-1) [x86_64]"
15:39:26.228 DEBUG main:228                         Style: "fusion"
15:39:26.228 DEBUG Lazy<T>:67                       DeviceFinders(0x55c56b0ae220) created
15:39:26.232 DEBUG Application:276                  Database(0x55c56b0a7560) moved to thread QThread(0x55c56b0f8c70)
15:39:26.232 DEBUG Lazy<T>:67                       Database(0x55c56b0a7560) created
15:39:26.232 DEBUG SCollection:70                   CollectionBackend(0x55c56b0a2d00) moved to thread QThread(0x55c56b0f8c70)
15:39:26.232 DEBUG Lazy<T>:67                       TaskManager(0x55c56b15cb50) created
15:39:26.233 DEBUG Application:276                  AlbumCoverLoader(0x55c56b17d5f0) moved to thread QThread(0x55c56b1809f0)
15:39:26.233 DEBUG Lazy<T>:67                       AlbumCoverLoader(0x55c56b17d5f0) created
15:39:26.234 DEBUG Lazy<T>:67                       SCollection(0x55c56b0af300) created
15:39:26.234 DEBUG SCollection:102                  CollectionWatcher(0x55c56b19b260) moved to thread Thread(0x55c56b191a10)
15:39:26.234 DEBUG Lazy<T>:67                       NetworkAccessManager(0x55c56b185140) created
15:39:26.234 DEBUG Lazy<T>:67                       LastFMImport(0x55c56b18ed90) created
15:39:26.234 DEBUG Application:276                  TagReaderClient(0x55c56b19a3c0) moved to thread QThread(0x55c56b19b1c0)
15:39:26.234 DEBUG Lazy<T>:67                       TagReaderClient(0x55c56b19a3c0) created
15:39:26.234 DEBUG NetworkProxyFactory:52           Detected system proxy URLs: QList("", "", "", "")
15:39:26.234 DEBUG WorkerPool<HandlerType>:257      Using worker "strawberry-tagreader" from "/home/stickman/Desktop/Strawberry_Test/strawberry/build"
15:39:26.234 DEBUG WorkerPool<HandlerType>:304      Starting worker 0x7fd23cde4220 "/home/stickman/Desktop/Strawberry_Test/strawberry/build/strawberry-tagreader" "/tmp/strawberry_1161866600"
15:39:26.236 DEBUG EngineBase:204                   SOUP_FORCE_HTTP1: ON
15:39:26.236 DEBUG Lazy<T>:67                       Player(0x55c56b0fa6b0) created
15:39:26.236 DEBUG Lazy<T>:67                       PlaylistManager(0x55c56b1db020) created
15:39:26.237 DEBUG Lazy<T>:67                       CurrentAlbumCoverLoader(0x55c56b1da440) created
15:39:26.240 WARN  GlobalShortcutsManager:206       No global shortcuts enabled.
15:39:26.251 INFO  main:47                          TagReader worker connecting to "/tmp/strawberry_1161866600"
15:39:26.251 DEBUG WorkerPool<HandlerType>:326      Worker 0x7fd21c001bc0 connected to "/tmp/strawberry_1161866600"
15:39:26.310 INFO  Player:954                       Registered URL handler for "subsonic"
15:39:26.313 DEBUG InternetServices:49              Added internet service "Subsonic"
15:39:26.313 INFO  Player:954                       Registered URL handler for "tidal"
15:39:26.321 DEBUG InternetServices:49              Added internet service "Tidal"
15:39:26.321 INFO  Player:954                       Registered URL handler for "qobuz"
15:39:26.329 DEBUG InternetServices:49              Added internet service "Qobuz"
15:39:26.329 DEBUG Lazy<T>:67                       InternetServices(0x55c56b0ad9c0) created
15:39:26.462 DEBUG MainWindow:349                   Starting
15:39:26.488 DEBUG CoverProviders:105               Registered cover provider "Last.fm"
15:39:26.488 DEBUG CoverProviders:105               Registered cover provider "MusicBrainz"
15:39:26.488 DEBUG CoverProviders:105               Registered cover provider "Discogs"
15:39:26.488 DEBUG CoverProviders:105               Registered cover provider "Deezer"
15:39:26.488 DEBUG CoverProviders:105               Registered cover provider "Musixmatch"
15:39:26.488 DEBUG CoverProviders:105               Registered cover provider "Spotify"
15:39:26.488 DEBUG CoverProviders:105               Registered cover provider "Tidal"
15:39:26.488 DEBUG CoverProviders:105               Registered cover provider "Qobuz"
15:39:26.488 DEBUG Lazy<T>:67                       CoverProviders(0x55c56c22a6b0) created
15:39:26.491 DEBUG LyricsProviders:104              Registered lyrics provider "Genius"
15:39:26.491 DEBUG LyricsProviders:104              Registered lyrics provider "Lyrics.ovh"
15:39:26.491 DEBUG LyricsProviders:104              Registered lyrics provider "LoloLyrics"
15:39:26.491 DEBUG LyricsProviders:104              Registered lyrics provider "Musixmatch"
15:39:26.491 DEBUG LyricsProviders:104              Registered lyrics provider "ChartLyrics"
15:39:26.491 DEBUG LyricsProviders:104              Registered lyrics provider "songlyrics.com"
15:39:26.491 DEBUG LyricsProviders:104              Registered lyrics provider "azlyrics.com"
15:39:26.491 DEBUG LyricsProviders:104              Registered lyrics provider "elyrics.net"
15:39:26.491 DEBUG LyricsProviders:104              Registered lyrics provider "lyricsmode.com"
15:39:26.491 DEBUG Lazy<T>:67                       LyricsProviders(0x55c56c453970) created
15:39:26.498 DEBUG MainWindow:400                   Initializing player
15:39:26.498 DEBUG MainWindow:410                   Creating models
15:39:26.498 DEBUG MainWindow:417                   Creating models finished
15:39:26.498 DEBUG DeviceLister:67                  CddaLister(0x55c56c6dfc00) moved to thread QThread(0x55c56c6dff80)
15:39:26.499 DEBUG DeviceLister:67                  Udisks2Lister(0x55c56c6e8370) moved to thread QThread(0x55c56c6e0040)
15:39:26.499 DEBUG DeviceLister:67                  GioLister(0x55c56c6e8eb0) moved to thread QThread(0x55c56c6e9af0)
15:39:26.499 DEBUG Lazy<T>:67                       DeviceManager(0x55c56c6df9f0) created
15:39:26.500 DEBUG logging:95(GLib-GIO)             Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
15:39:26.501 DEBUG Application:276                  PlaylistBackend(0x55c56c6ef260) moved to thread QThread(0x55c56b0f8c70)
15:39:26.501 DEBUG Lazy<T>:67                       PlaylistBackend(0x55c56c6ef260) created
15:39:26.501 DEBUG logging:95(GLib-GIO)             Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
15:39:26.502 DEBUG logging:95(GLib-GIO)             Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
15:39:26.504 DEBUG logging:95(GLib-GIO)             _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
15:39:26.575 DEBUG Lazy<T>:67                       OrganizeDialog(0x55c56c72f2c0, name="OrganizeDialog") created
15:39:26.576 DEBUG Application:276                  RadioBackend(0x55c56c7e4970) moved to thread QThread(0x55c56b0f8c70)
15:39:26.576 DEBUG RadioServices:66                 Adding radio service: "SomaFM"
15:39:26.576 DEBUG RadioServices:66                 Adding radio service: "Radio Paradise"
15:39:26.576 DEBUG Lazy<T>:67                       RadioServices(0x55c56c738c60) created
15:39:26.576 DEBUG MainWindow:437                   Creating UI
15:39:26.590 DEBUG AudioScrobbler:63                Registered scrobbler service "Last.fm"
15:39:26.590 DEBUG AudioScrobbler:63                Registered scrobbler service "Libre.fm"
15:39:26.590 DEBUG AudioScrobbler:63                Registered scrobbler service "ListenBrainz"
15:39:26.590 DEBUG AudioScrobbler:63                Registered scrobbler service "Subsonic"
15:39:26.590 DEBUG Lazy<T>:67                       AudioScrobbler(0x55c56cb4ea50) created
15:39:26.602 DEBUG Lazy<T>:67                       MoodbarController(0x55c56cd24330) created
15:39:26.602 DEBUG MainWindow:860                   Creating playing widget
15:39:26.618 DEBUG MainWindow:907                   Loading settings
15:39:26.653 DEBUG MainWindow:1072                  Started QThread(0x55c56b038df0)
15:39:26.657 INFO  DeviceManager:450                Device added: "/dev/sr0" QList("/dev/sr0")
15:39:26.659 INFO  DeviceManager:450                Device added: "Gio/unmounted/140539450287040" QList("Gio//smb2/3933013655552", "Gio/unmounted/140539450287040")
15:39:26.660 INFO  DeviceManager:450                Device added: "Gio//smb2/3933013655552" QList("Gio//smb2/3933013655552", "Gio/unmounted/140539450287040")
15:39:26.663 INFO  DeviceManager:450                Device added: "Gio//smb2/3933013655552" QList("Gio//smb2/3933013655552", "Gio/unmounted/140539450287040")
15:39:26.663 INFO  DeviceManager:450                Device added: "Gio//smb2/3933013655552" QList("Gio//smb2/3933013655552", "Gio/unmounted/140539450287040")
15:39:26.663 INFO  DeviceManager:450                Device added: "Gio//smb2/3933013655552" QList("Gio//smb2/3933013655552", "Gio/unmounted/140539450287040")
15:39:26.663 INFO  DeviceManager:450                Device added: "Gio//smb2/3933013655552" QList("Gio//smb2/3933013655552", "Gio/unmounted/140539450287040")
15:39:26.663 INFO  DeviceManager:450                Device added: "Gio//smb2/3933013655552" QList("Gio//smb2/3933013655552", "Gio/unmounted/140539450287040")
15:39:27.536 DEBUG PlaylistBackend:333              Saving playlist 1
15:39:55.864 DEBUG Database:485                     Starting database integrity check
15:39:55.925 DEBUG Database:555                     Starting database backup
15:40:02.989 DEBUG _MessageReplyBase:31             Waiting on ID 0
15:40:02.989 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/01 Wesley’s Theory (Feat George Clinton & Thundercat).opus"
15:40:03.010 DEBUG MessageReply<MessageType>:90     Releasing ID 0 (finished)
15:40:03.010 DEBUG _MessageReplyBase:33             Acquired ID 0
15:40:03.010 DEBUG _MessageReplyBase:31             Waiting on ID 1
15:40:03.010 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/02 For Free (Interlude).opus"
15:40:03.040 DEBUG MessageReply<MessageType>:90     Releasing ID 1 (finished)
15:40:03.040 DEBUG _MessageReplyBase:33             Acquired ID 1
15:40:03.040 DEBUG _MessageReplyBase:31             Waiting on ID 2
15:40:03.040 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/03 King Kunta.opus"
15:40:03.062 DEBUG MessageReply<MessageType>:90     Releasing ID 2 (finished)
15:40:03.062 DEBUG _MessageReplyBase:33             Acquired ID 2
15:40:03.062 DEBUG _MessageReplyBase:31             Waiting on ID 3
15:40:03.062 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/04 Institutionalized (Feat Bilal, Anna Wise & Snoop Dogg).opus"
15:40:03.079 DEBUG MessageReply<MessageType>:90     Releasing ID 3 (finished)
15:40:03.080 DEBUG _MessageReplyBase:33             Acquired ID 3
15:40:03.080 DEBUG _MessageReplyBase:31             Waiting on ID 4
15:40:03.080 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/05 These Walls (Feat Bilal, Anna Wise & Thundercat).opus"
15:40:03.099 DEBUG MessageReply<MessageType>:90     Releasing ID 4 (finished)
15:40:03.099 DEBUG _MessageReplyBase:33             Acquired ID 4
15:40:03.099 DEBUG _MessageReplyBase:31             Waiting on ID 5
15:40:03.099 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/06 U.opus"
15:40:03.116 DEBUG MessageReply<MessageType>:90     Releasing ID 5 (finished)
15:40:03.116 DEBUG _MessageReplyBase:33             Acquired ID 5
15:40:03.116 DEBUG _MessageReplyBase:31             Waiting on ID 6
15:40:03.116 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/07 Alright.opus"
15:40:03.143 DEBUG MessageReply<MessageType>:90     Releasing ID 6 (finished)
15:40:03.144 DEBUG _MessageReplyBase:33             Acquired ID 6
15:40:03.144 DEBUG _MessageReplyBase:31             Waiting on ID 7
15:40:03.144 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/08 For Sale (Interlude).opus"
15:40:03.173 DEBUG MessageReply<MessageType>:90     Releasing ID 7 (finished)
15:40:03.173 DEBUG _MessageReplyBase:33             Acquired ID 7
15:40:03.173 DEBUG _MessageReplyBase:31             Waiting on ID 8
15:40:03.173 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/09 Momma.opus"
15:40:03.198 DEBUG MessageReply<MessageType>:90     Releasing ID 8 (finished)
15:40:03.198 DEBUG _MessageReplyBase:33             Acquired ID 8
15:40:03.199 DEBUG _MessageReplyBase:31             Waiting on ID 9
15:40:03.199 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/10 Hood Politics.opus"
15:40:03.215 DEBUG MessageReply<MessageType>:90     Releasing ID 9 (finished)
15:40:03.215 DEBUG _MessageReplyBase:33             Acquired ID 9
15:40:03.215 DEBUG _MessageReplyBase:31             Waiting on ID 10
15:40:03.215 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/11 How Much A Dollar Cost (Feat James Fauntleroy & Ronald Isley).opus"
15:40:03.237 DEBUG MessageReply<MessageType>:90     Releasing ID 10 (finished)
15:40:03.237 DEBUG _MessageReplyBase:33             Acquired ID 10
15:40:03.237 DEBUG _MessageReplyBase:31             Waiting on ID 11
15:40:03.237 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/12 Complexion (A Zulu Love) (Feat Rapsody).opus"
15:40:03.261 DEBUG MessageReply<MessageType>:90     Releasing ID 11 (finished)
15:40:03.261 DEBUG _MessageReplyBase:33             Acquired ID 11
15:40:03.261 DEBUG _MessageReplyBase:31             Waiting on ID 12
15:40:03.261 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/13 The Blacker The Berry.opus"
15:40:03.284 DEBUG MessageReply<MessageType>:90     Releasing ID 12 (finished)
15:40:03.284 DEBUG _MessageReplyBase:33             Acquired ID 12
15:40:03.284 DEBUG _MessageReplyBase:31             Waiting on ID 13
15:40:03.284 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/14 You Ain’t Gotta Lie (Momma Said).opus"
15:40:03.305 DEBUG MessageReply<MessageType>:90     Releasing ID 13 (finished)
15:40:03.305 DEBUG _MessageReplyBase:33             Acquired ID 13
15:40:03.305 DEBUG _MessageReplyBase:31             Waiting on ID 14
15:40:03.305 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/15 I.opus"
15:40:03.332 DEBUG MessageReply<MessageType>:90     Releasing ID 14 (finished)
15:40:03.332 DEBUG _MessageReplyBase:33             Acquired ID 14
15:40:03.332 DEBUG _MessageReplyBase:31             Waiting on ID 15
15:40:03.332 DEBUG TagReaderTagLib:237              Reading tags from "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/16 Mortal Man.opus"
15:40:03.354 DEBUG MessageReply<MessageType>:90     Releasing ID 15 (finished)
15:40:03.354 DEBUG _MessageReplyBase:33             Acquired ID 15
15:40:06.838 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/01 Wesley’s Theory (Feat George Clinton & Thundercat).opus"
15:40:06.855 DEBUG CollectionWatcher:1068           Subdir "/media/MyMedia/MUSIC/Kendrick Lamar" changed under directory "/media/MyMedia/MUSIC" id 1
15:40:06.855 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/02 For Free (Interlude).opus"
15:40:06.855 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/03 King Kunta.opus"
15:40:06.855 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/04 Institutionalized (Feat Bilal, Anna Wise & Snoop Dogg).opus"
15:40:06.855 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/05 These Walls (Feat Bilal, Anna Wise & Thundercat).opus"
15:40:06.856 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/06 U.opus"
15:40:06.856 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/07 Alright.opus"
15:40:06.856 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/08 For Sale (Interlude).opus"
15:40:06.856 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/09 Momma.opus"
15:40:06.856 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/10 Hood Politics.opus"
15:40:06.951 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/11 How Much A Dollar Cost (Feat James Fauntleroy & Ronald Isley).opus"
15:40:06.951 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/12 Complexion (A Zulu Love) (Feat Rapsody).opus"
15:40:06.951 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/13 The Blacker The Berry.opus"
15:40:06.951 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/14 You Ain’t Gotta Lie (Momma Said).opus"
15:40:06.951 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/15 I.opus"
15:40:06.951 INFO  Organize:173                     Processing "/media/MyMedia/music to add/!UPGRADE/To Pimp A Butterfly/16 Mortal Man.opus"
15:40:09.960 DEBUG _MessageReplyBase:31             Waiting on ID 16
15:40:09.960 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/11 How Much A Dollar Cost (Feat James Fauntleroy & Ronald Isley).opus"
15:40:09.961 DEBUG MessageReply<MessageType>:90     Releasing ID 16 (finished)
15:40:09.961 DEBUG _MessageReplyBase:33             Acquired ID 16
15:40:09.961 DEBUG _MessageReplyBase:31             Waiting on ID 17
15:40:09.961 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/07 Alright.opus"
15:40:09.961 DEBUG MessageReply<MessageType>:90     Releasing ID 17 (finished)
15:40:09.961 DEBUG _MessageReplyBase:33             Acquired ID 17
15:40:09.961 DEBUG _MessageReplyBase:31             Waiting on ID 18
15:40:09.961 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/01 Wesley’s Theory (Feat George Clinton & Thundercat).opus"
15:40:09.961 DEBUG MessageReply<MessageType>:90     Releasing ID 18 (finished)
15:40:09.961 DEBUG _MessageReplyBase:33             Acquired ID 18
15:40:09.961 DEBUG _MessageReplyBase:31             Waiting on ID 19
15:40:09.961 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/09 Momma.opus"
15:40:09.961 DEBUG MessageReply<MessageType>:90     Releasing ID 19 (finished)
15:40:09.961 DEBUG _MessageReplyBase:33             Acquired ID 19
15:40:09.961 DEBUG _MessageReplyBase:31             Waiting on ID 20
15:40:09.961 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/13 The Blacker The Berry.opus"
15:40:09.961 DEBUG MessageReply<MessageType>:90     Releasing ID 20 (finished)
15:40:09.961 DEBUG _MessageReplyBase:33             Acquired ID 20
15:40:09.961 DEBUG _MessageReplyBase:31             Waiting on ID 21
15:40:09.961 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/04 Institutionalized (Feat Bilal, Anna Wise & Snoop Dogg).opus"
15:40:09.961 DEBUG MessageReply<MessageType>:90     Releasing ID 21 (finished)
15:40:09.961 DEBUG _MessageReplyBase:33             Acquired ID 21
15:40:09.961 DEBUG _MessageReplyBase:31             Waiting on ID 22
15:40:09.961 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/10 Hood Politics.opus"
15:40:09.961 DEBUG MessageReply<MessageType>:90     Releasing ID 22 (finished)
15:40:09.961 DEBUG _MessageReplyBase:33             Acquired ID 22
15:40:09.961 DEBUG _MessageReplyBase:31             Waiting on ID 23
15:40:09.961 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/16 Mortal Man.opus"
15:40:09.961 DEBUG MessageReply<MessageType>:90     Releasing ID 23 (finished)
15:40:09.961 DEBUG _MessageReplyBase:33             Acquired ID 23
15:40:09.961 DEBUG _MessageReplyBase:31             Waiting on ID 24
15:40:09.961 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/14 You Ain’t Gotta Lie (Momma Said).opus"
15:40:09.961 DEBUG MessageReply<MessageType>:90     Releasing ID 24 (finished)
15:40:09.961 DEBUG _MessageReplyBase:33             Acquired ID 24
15:40:09.962 DEBUG _MessageReplyBase:31             Waiting on ID 25
15:40:09.962 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/06 U.opus"
15:40:09.962 DEBUG MessageReply<MessageType>:90     Releasing ID 25 (finished)
15:40:09.962 DEBUG _MessageReplyBase:33             Acquired ID 25
15:40:09.962 DEBUG _MessageReplyBase:31             Waiting on ID 26
15:40:09.962 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/08 For Sale (Interlude).opus"
15:40:09.962 DEBUG MessageReply<MessageType>:90     Releasing ID 26 (finished)
15:40:09.962 DEBUG _MessageReplyBase:33             Acquired ID 26
15:40:09.962 DEBUG _MessageReplyBase:31             Waiting on ID 27
15:40:09.962 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/02 For Free (Interlude).opus"
15:40:09.962 DEBUG MessageReply<MessageType>:90     Releasing ID 27 (finished)
15:40:09.962 DEBUG _MessageReplyBase:33             Acquired ID 27
15:40:09.962 DEBUG _MessageReplyBase:31             Waiting on ID 28
15:40:09.962 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/05 These Walls (Feat Bilal, Anna Wise & Thundercat).opus"
15:40:09.962 DEBUG MessageReply<MessageType>:90     Releasing ID 28 (finished)
15:40:09.962 DEBUG _MessageReplyBase:33             Acquired ID 28
15:40:09.962 DEBUG _MessageReplyBase:31             Waiting on ID 29
15:40:09.962 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/03 King Kunta.opus"
15:40:09.962 DEBUG MessageReply<MessageType>:90     Releasing ID 29 (finished)
15:40:09.962 DEBUG _MessageReplyBase:33             Acquired ID 29
15:40:09.962 DEBUG _MessageReplyBase:31             Waiting on ID 30
15:40:09.962 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/15 I.opus"
15:40:09.962 DEBUG MessageReply<MessageType>:90     Releasing ID 30 (finished)
15:40:09.962 DEBUG _MessageReplyBase:33             Acquired ID 30
15:40:09.962 DEBUG _MessageReplyBase:31             Waiting on ID 31
15:40:09.962 DEBUG TagReaderTagLib:194              Checking for valid file "/media/MyMedia/MUSIC/Kendrick Lamar/To Pimp A Butterfly/12 Complexion (A Zulu Love) (Feat Rapsody).opus"
15:40:09.962 DEBUG MessageReply<MessageType>:90     Releasing ID 31 (finished)
15:40:09.962 DEBUG _MessageReplyBase:33             Acquired ID 31
`

**Comment by jonaski** [comment]
Sorry, I misread the source path, now I see that it is "/media/MyMedia/music to add/!UPGRADE".
The pasted logs is a bit hard to read since all newlines are stripped, I've seen that happen to me before too, maybe it's the specific terminal program that does that, I don't remember, maybe try copying to text editor first. I use Konsole now and it does not do that.

I see now that there is a case where nothing is logged, that's when the song already exists in the DB but is marked unavailable and rediscovered. So if you are testing with the same songs and the same paths, there are probably still duplicates in the DB except the songs are marked unavailable so they do not appear in the collection, so even if I fixed the bug the duplicates can reappear when you organize, so you need to clear all the duplicates in the database. The easiest is just to clear all unavailable songs, which are not useful anyway unless you actually have a path/disk that is currently unavailable and you plan to restore the same songs on the same paths.

Make a copy of ~/.local/share/strawberry/strawberry/strawberry.db just in case.

You can list the unavailable song with:

    sqlite3 ~/.local/share/strawberry/strawberry/strawberry.db
    select url from songs where unavailable = 1;

If the songs listed there are safe to remove from the database, you can delete all unavailable songs with the following SQL queries:

    delete from songs where unavailable = 1;

Recreate FTS table:

    delete from strawberry.songs_fts;
    insert into songs_fts (ROWID, ftstitle, ftsalbum, ftsartist, ftsalbumartist, ftscomposer, ftsperformer, ftsgrouping, ftsgenre, 
ftscomment) select ROWID, title, album, artist, albumartist, composer, performer, grouping, genre, comment FROM songs;

**Comment by manofsticks** [comment]
That looks like it fixed it! Awesome, thank you very much! That problem makes sense, as my typical "workflow" was to delete the folder from the filesystem via a file browser, not from Strawberry. I had an enormous amount of "Unavailable" tracks in there.

The "Recreate FTS table" SQL you provided was slightly off with a typo (mostly commenting this in case anyone comes across this and is trying to copy the fix). The SQL I used was:

```
delete from songs_fts;

insert into songs_fts (ROWID, ftstitle, ftsalbum, ftsartist, ftsalbumartist, ftscomposer, ftsperformer, ftsgrouping, ftsgenre, ftscomment) select ROWID, title, album, artist, albumartist, composer, performer, grouping, genre, comment FROM songs;
```
Also I've used Terminator for my emulator; will make sure I get newline working before future bug reports.

**Comment by manofsticks** [comment]
I've just run into this issue again with Strawberry 1.1.1. 

Not sure how much this helps, but checking the songs database shows 2 records, neither are "unavailable", and both point to the same file (I can play either one from the GUI). Same thing for track 2. 

![image](https://github.com/user-attachments/assets/47c3f049-acd4-4ff4-8c05-967ba28ae5aa)

![image](https://github.com/user-attachments/assets/9c0a24a9-63b9-4608-b3d9-c705ac771d94)

Here's regular log

`[stickman@Sweden /home/stickman]$ strawberry --verbose
10:47:16.925 INFO  main:181                         Strawberry 1.1.1 Qt "6.7.2"
10:47:16.925 INFO  main:182                         "Arch Linux unknown - (linux 6.10.6-arch1-1) [x86_64]"
10:47:16.943 WARN  GlobalShortcutsManager:207       No global shortcuts enabled.
10:47:16.949 INFO  main:47                          TagReader worker connecting to "/tmp/strawberry_188227867"
10:47:17.030 INFO  Player:973                       Registered URL handler for "subsonic"
10:47:17.031 INFO  Player:973                       Registered URL handler for "tidal"
10:47:17.039 INFO  Player:973                       Registered URL handler for "qobuz"
10:47:17.367 INFO  DeviceManager:450                Device added: "Gio/unmounted/140345371419664" QList("Gio/unmounted/140345371416432", "Gio/unmounted/140345371417920", "Gio/unmounted/140345371419664")
10:47:17.368 INFO  DeviceManager:450                Device added: "Gio/unmounted/140345371417920" QList("Gio/unmounted/140345371416432", "Gio/unmounted/140345371417920", "Gio/unmounted/140345371419664")
10:47:17.368 INFO  DeviceManager:450                Device added: "Gio/unmounted/140345371416432" QList("Gio/unmounted/140345371416432", "Gio/unmounted/140345371417920", "Gio/unmounted/140345371419664")
10:47:17.370 ERROR TagReaderTagLib:314              File "/media/MyMedia/MUSIC/Flo Rida/Wild Ones/02 Wild Ones.opus" does not exist
10:47:17.370 ERROR CollectionPlaylistItem:46        Could not reload file QUrl("file:///media/MyMedia/MUSIC/Flo Rida/Wild Ones/02 Wild Ones.opus") "File does not exist"
10:47:17.370 ERROR TagReaderTagLib:314              File "/media/MyMedia/MUSIC/Flo Rida/Wild Ones/01 Whistle.opus" does not exist
10:47:17.370 ERROR CollectionPlaylistItem:46        Could not reload file QUrl("file:///media/MyMedia/MUSIC/Flo Rida/Wild Ones/01 Whistle.opus") "File does not exist"
10:47:17.370 ERROR TagReaderTagLib:314              File "/media/MyMedia/MUSIC/Flo Rida/Wild Ones/01 Whistle.opus" does not exist
10:47:17.370 ERROR CollectionPlaylistItem:46        Could not reload file QUrl("file:///media/MyMedia/MUSIC/Flo Rida/Wild Ones/01 Whistle.opus") "File does not exist"
10:47:17.370 ERROR TagReaderTagLib:314              File "/media/MyMedia/MUSIC/Flo Rida/Wild Ones/02 Wild Ones.opus" does not exist
10:47:17.370 ERROR CollectionPlaylistItem:46        Could not reload file QUrl("file:///media/MyMedia/MUSIC/Flo Rida/Wild Ones/02 Wild Ones.opus") "File does not exist"
10:47:17.370 ERROR TagReaderTagLib:314              File "/media/MyMedia/MUSIC/Flo Rida/Wild Ones/02 Wild Ones.opus" does not exist
10:47:17.370 ERROR CollectionPlaylistItem:46        Could not reload file QUrl("file:///media/MyMedia/MUSIC/Flo Rida/Wild Ones/02 Wild Ones.opus") "File does not exist"
10:47:17.749 INFO  DeviceManager:450                Device added: "/dev/sr0" QList("/dev/sr0")
10:47:26.961 INFO  Organize:177                     Processing "/media/MyMedia/music to add/Wild Ones/01 Whistle.opus"
10:47:26.962 INFO  Organize:177                     Processing "/media/MyMedia/music to add/Wild Ones/02 Wild Ones.opus"
10:47:26.962 INFO  Organize:177                     Processing "/media/MyMedia/music to add/Wild Ones/03 Let It Roll.opus"
10:47:26.962 INFO  Organize:177                     Processing "/media/MyMedia/music to add/Wild Ones/04 Good Feeling.opus"
10:47:26.962 INFO  Organize:177                     Processing "/media/MyMedia/music to add/Wild Ones/05 In My Mind, Part 2.opus"
10:47:26.962 INFO  Organize:177                     Processing "/media/MyMedia/music to add/Wild Ones/06 Sweet Spot.opus"
10:47:26.962 INFO  Organize:177                     Processing "/media/MyMedia/music to add/Wild Ones/07 Thinking of You.opus"
10:47:26.962 INFO  Organize:177                     Processing "/media/MyMedia/music to add/Wild Ones/08 I Cry.opus"
10:47:26.962 INFO  Organize:177                     Processing "/media/MyMedia/music to add/Wild Ones/09 Run.opus"

`

**Comment by spirillen** [comment]
I to have the similar issue on ubuntu 24.04 STD repo release. Even when I do delete the DB, remove the source, and re add it, all songs are duplicated in the collection tree, This goes for all medias. (flack, m4a mostly)

![image](https://github.com/user-attachments/assets/64b5fd3c-5209-42a9-85b6-660f29357223)

**Comment by WildPenquin** [comment]
I also have duplicate tracks of a great portion of my library.

It seems that all duplicates are different formats of the same track, even when I only have one file. Say, if I have only an OGG Vorbis file, there may be duplicates of OGG, MP3 or FLAC in the db - but the other files are actually non-existent. Also, Playing the MP3 version (as shown by File Type) actually plays the OGG version, and playing the OGG version will fail with an error as Strawberry can not find the file, but it lists the file it can not find in the error dialog pointing to the OGG file I actually have on the disk.

I tried to move away all versions of a single album which has this issue, and did a full collection re-scan. These "ghost files" will still remain in the collection (and can not be played, verifying they are not stray files I'd have misplaced somehow).

I will be resetting the collection db (moving away, deleting it), however as I don't know why it happened in the first place, only time will tell if it will re-occur.

I do re-tag my files often with Picard and I am in the process of re-encoding originals as FLAC gradually. It is possible Strawberry has been confused by some renaming and/or deletion of files (however that should still of course not happen).

**EDIT:** Seems like moving away `~/.local/share/strawberry/strawberry/strawberry.db` will also delete all my playlists, too. Is there a way to actually do a full re-scan / wipe of the whole collection without removing playlists?

**Comment by elnull** [comment]
Duplicates also appears after updating collection with `beet mbsync` command and full rescan in Strawberry.
